src/background/services/tabState.ts

```diff
@@ -0,0 +1,60 @@
+import { Tabs } from 'webextension-polyfill'
+
+type State = {
+  lastPaymentTimestamp: number
+  expiresAtTimestamp: number
+}
+
+export class TabState {
+  private state = new WeakMap<Tabs.Tab, Map<string, State>>()
+
+  constructor() {}
+
+  private getStateKey(url: string, walletAddressId: string): string {
```

Comment:
Keeping https://github.com/interledger/web-monetization-extension/issues/381 in mind, maybe should change current `this.state` to something like `this.lastPaymentInfoState` (as the keys could be different)?

```suggestion
  private getOverpayingStateKey(url: string, walletAddressId: string): string {
```